knowledge_verificator/command_line.py

```diff
@@ -0,0 +1,135 @@
+            answer provided by a user. Either they are consistent, not
+            consistent or they are independent claims.
+        chosen_answer (str): An answer provided by a user.
+    """
+    match relation:
+        case Relation.ENTAILMENT:
+            feedback = 'correct'
+            style = 'green'
+        case Relation.CONTRADICTION:
+            feedback = f'wrong. Correct answer is {chosen_answer}'
+            style = 'red'
+        case Relation.NEUTRAL:
+            feedback = 'not directly associated with the posed question'
+            style = 'yellow'
+
+    feedback_text = Text(f'Your answer is {feedback}.', style=style)
+    console.print(feedback_text)
+
+
+def run_cli_mode():
+    """
+    Run an interactive command-line interface.
+
+    Raises:
+        ValueError:
+    """
+    qg_module = QuestionGeneration()
+    chooser = AnswerChooser()
+    nli_module = NaturalLanguageInference()
+
+    while True:
+        options = ['knowledge database', 'my own paragraph']
+        user_choice = choose_from_menu(
+            menu_elements=options, plural_name='options'
+        )
+
+        match user_choice:
+            case 'knowledge database':
+                try:
+                    material_db = MaterialDatabase(config.learning_materials)
+                except FileNotFoundError:
+                    console.print(
+                        f'In the `{config.learning_materials}` there is no database. '
+                        'Try using your own materials.'
+                    )
+                    continue
+
+                if not material_db.materials:
+                    console.print(
+                        'The knowledge database exists but is empty. '
+                        'Try using your own materials.'
+                    )
+                    continue
+
+                material = choose_from_menu(
+                    material_db.materials,
+                    plural_name='materials',
+                    attribute_to_show='title',
+                )
+
+                if material is None:
+                    continue
+
+                paragraph = str(
+                    choose_from_menu(material.paragraphs, 'paragraphs')
+                )
+
+                if paragraph is None:
+                    continue
+
+                console.print('Learn this paragraph: ')
+                console.print(paragraph)
+                console.print()
+                input('Press ENTER when ready.')
+
+            case 'my own paragraph':
+                console.print('Enter a paragraph you would like to learn: ')
+                paragraph = input().strip()
+
+            case _:
+                console.print('Unrecognised option, try again!')
+
+        logger.debug('Loaded the following paragraph:\n %s', paragraph)
+
+        chosen_answer = chooser.choose_answer(paragraph=paragraph)
+        if not chosen_answer:
+            logger.error(
```

Comment:
`''knowledge database''` section uses `console.print` for errors, if its fine because
in `"knowledge database"` error was directly caused by user input, and here you are just loading some data, why not move this check to the paragraph loading code, and make sure all available paragraphs are good?

Reply:
Good point.